app/controllers/think_feel_do_dashboard/reports_controller.rb

```diff
@@ -3,64 +3,29 @@
 module ThinkFeelDoDashboard
   # Provide access to data exports.
   class ReportsController < ApplicationController
-    REPORTS = [
-      {
-        title: "Site Session",
-        id: :site_session,
-        klass: Reports::SiteSession
-      },
-      {
-        title: "Task Completion",
-        id: :task_completion,
-        klass: Reports::TaskCompletion
-      },
-      {
-        title: "Tool Access",
-        id: :tool_access,
-        klass: Reports::ToolAccess
-      },
-      {
-        title: "User Agent",
-        id: :user_agent,
-        klass: Reports::UserAgent
-      },
-      {
-        title: "Login",
-        id: :login,
-        klass: Reports::Login
-      },
-      {
-        title: "PHQ9 Assessment",
-        id: :phq_assessment,
-        klass: Reports::PhqAssessment
-      },
-      {
-        title: "WAI Assessment",
-        id: :wai_assessment,
-        klass: Reports::WaiAssessment
-      }
-    ]
-
     def index
-      authorize! :read, Reports::SiteSession
-      reports
+      authorize! :read, "Reports"
+
+      fetch_reports
     end
 
     def show
-      report = (reports.find { |r| r[:id].to_s == params[:id] } || {})[:klass]
+      authorize! :read, "Reports"
+
+      report = fetch_reports.find { |r| r[:id].to_s == params[:id] } || {}
+      report = report[:klass]
 
-      authorize! :download, report
       respond_to { |fmt| fmt.csv { send_data report.to_csv, type: "text/csv" } }
     end
 
     private
 
-    def reports
-      @reports = REPORTS
+    def fetch_reports
+      @reports = []
 
       return @reports unless Rails.application.config.respond_to?(:reports)
 
-      @reports += Rails.application.config.reports
+      @reports = Rails.application.config.reports
```

Comment:
See https://github.com/cbitstech/think_feel_do/blob/master/config/initializers/think_feel_do.rb


-------- Original message --------
From: Gwen Smuda <notifications@github.com>
Date:02/24/2015 3:08 PM (GMT-06:00)
To: cbitstech/think_feel_do_dashboard <think_feel_do_dashboard@noreply.github.com>
Cc: Eric M Carty Fickes <ericcf@northwestern.edu>
Subject: Re: [think_feel_do_dashboard] Ecf reports (#10)


In app/controllers/think_feel_do_dashboard/reports_controller.rb<https://github.com/cbitstech/think_feel_do_dashboard/pull/10#discussion_r25292637>:

>
>        return @reports unless Rails.application.config.respond_to?(:reports)
>
> -      @reports += Rails.application.config.reports
> +      @reports = Rails.application.config.reports


So, the particular kinds of a reports available to an application will be set in a host app configuration as an array of report klasses?

Is this implemented yet in either of the host apps? I didn't see this in the commits.

-
Reply to this email directly or view it on GitHub<https://github.com/cbitstech/think_feel_do_dashboard/pull/10/files#r25292637>.